src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -57,11 +103,48 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
+        public IEnumerable<ConfigurationClient> GetAvailableClients()
         {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
-        }
-
+            return _clients.Where(client => client.BackoffEndTime <= DateTimeOffset.UtcNow).Select(c => c.Client).ToList();
```

Comment:
Invoking DateTimeOffset.UtcNow in this predicate causes it to be evaluated each time. It should be evaluted once outside.

Reply:
I don't see any side effect of evaluating it each time,  remind me please if you have.

Reply:
the side-effect is that a client that got backed off can become eligible again. The current design is that we should make at most one round through clients.

Reply:
There's no side effect. It looks unintentional. It's better to avoid repeated invocation of UtcNow when unneeded.

Reply:
Oh, right, I misunderstanded it.